sycl/source/detail/scheduler/commands.cpp

```diff
@@ -403,9 +377,69 @@ void Command::makeTraceEventEpilog() {
 #endif
 }
 
+void Command::addDepSub(EventImplPtr DepEvent, ContextImplPtr Context) {
+  // Async work is not supported for host device.
+  if (DepEvent->is_host()) {
+    // call to waitInternal() is in prepareEvents() as it's called from
+    // enqueue process functions
+    return;
+  }
+
+  if (DepEvent->getHandleRef() == nullptr) {
+    return;
+  }
+
+  ContextImplPtr DepEventContext = DepEvent->getContextImpl();
+  // If contexts don't match - connect them using user event
+  if (DepEventContext != Context && !Context->is_host()) {
```

Comment:
Could you please simplify this and use internal API instead of external one.
Specifically I suggest to construct Host Task type command manually and make it depends on  `DepEvent`.

Reply:
Done.